Content.Server/GameObjects/Components/Interactable/HandheldLightComponent.cs

```diff
@@ -1,79 +1,84 @@
 using Content.Server.GameObjects.Components.Items.Storage;
 using Content.Server.GameObjects.Components.Power;
-using Content.Server.Interfaces.GameObjects.Components.Items;
 using Content.Shared.GameObjects.Components;
 using Content.Shared.GameObjects.EntitySystems;
-using Content.Shared.GameObjects.Verbs;
 using Content.Shared.Interfaces;
 using Content.Shared.Interfaces.GameObjects.Components;
 using Robust.Server.GameObjects;
-using Robust.Server.GameObjects.Components.Container;
 using Robust.Server.GameObjects.EntitySystems;
-using Robust.Server.Interfaces.GameObjects;
 using Robust.Shared.GameObjects;
 using Robust.Shared.GameObjects.Systems;
 using Robust.Shared.Interfaces.GameObjects;
 using Robust.Shared.Localization;
+using Robust.Shared.Serialization;
 using Robust.Shared.Utility;
 using Robust.Shared.ViewVariables;
 
 namespace Content.Server.GameObjects.Components.Interactable
 {
     /// <summary>
-    ///     Component that represents a handheld lightsource which can be toggled on and off.
+    ///     Component that represents a powered handheld light source which can be toggled on and off.
     /// </summary>
     [RegisterComponent]
-    internal sealed class HandheldLightComponent : SharedHandheldLightComponent, IUse, IExamine, IInteractUsing,
-        IMapInit
+    internal sealed class HandheldLightComponent : SharedHandheldLightComponent, IUse, IExamine, IInteractUsing
     {
-        [ViewVariables(VVAccess.ReadWrite)] public float Wattage { get; set; } = 10;
-        [ViewVariables] private ContainerSlot _cellContainer = default!;
-
-        [ViewVariables]
-        private BatteryComponent? Cell
-        {
-            get
-            {
-                if (_cellContainer.ContainedEntity == null) return null;
-                if (_cellContainer.ContainedEntity.TryGetComponent(out BatteryComponent? cell))
-                {
-                    return cell;
-                }
-
-                return null;
-            }
-        }
+        [ViewVariables(VVAccess.ReadWrite)] public float Wattage = 10f;
+        [ViewVariables] private PowerCellSlotComponent _cellSlot = default!;
+        private PowerCellComponent? Cell => _cellSlot.Cell;
 
         /// <summary>
         ///     Status of light, whether or not it is emitting light.
         /// </summary>
         [ViewVariables]
         public bool Activated { get; private set; }
 
-        [ViewVariables] protected override bool HasCell => Cell != null;
+        [ViewVariables] protected override bool HasCell => _cellSlot.HasCell;
 
-        async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
+        [ViewVariables(VVAccess.ReadWrite)] public string TurnOnSound = "/Audio/Items/flashlight_toggle.ogg";
+        [ViewVariables(VVAccess.ReadWrite)] public string TurnOnFailSound = "/Audio/Machines/button.ogg";
+        [ViewVariables(VVAccess.ReadWrite)] public string TurnOffSound = "/Audio/Items/flashlight_toggle.ogg";
```

Comment:
Redundant to specify it here and in ExposeData considering ExposeData is always called.

Reply:
I also like prefixing with sound rather than suffixing so it makes IDE autocompletion easier but that's just me.